public/app/plugins/datasource/grafana-azure-monitor-datasource/components/AnalyticsConfig.tsx

```diff
@@ -149,12 +133,12 @@ export class AnalyticsConfig extends PureComponent<Props, State> {
         <Switch
           label="Same details as Azure Monitor API"
           checked={jsonData.azureLogAnalyticsSameAs}
-          onChange={event => this.onAzureLogAnalyticsSameAsChange(!jsonData.azureLogAnalyticsSameAs)}
+          onChange={() => this.onAzureLogAnalyticsSameAsChange(!jsonData.azureLogAnalyticsSameAs)}
```

Comment:
Nitpick: maybe avoid anonymous functions.
```suggestion
          onChange={this.onAzureLogAnalyticsSameAsChange()}
```

Reply:
@hugohaggmark this will call `onAzurelLogAnalyticsSameAsChange` on every render. @shavonn you need just function reference here, not the call

Reply:
Sorry missed the `()` in my suggestion, updated it now.

Reply:
great!